tests/scan/test_scannable.py

```diff
@@ -179,3 +183,53 @@ def test_patch_max_size():
     files = list(c.get_files())
 
     assert len(files) == 0
+
+
+@pytest.mark.parametrize("absolute_path", (True, False))
+def test_get_tar_stream(tmp_path, absolute_path):
+    """
+    GIVEN a Files object, representing paths, either absolute or relative
+    WHEN the get_tar_stream method is called
+    THEN a BytesIO stream is outputted, representing a tar of the files represented by the Files object
+    """
+    file1_path = tmp_path / "file1.txt" if absolute_path else Path("file1.txt")
+    dir_path = tmp_path / "my_test_dir" if absolute_path else Path("my_test_dir")
+    file2_path = dir_path / "file2.txt"
+    file1_content = "My first document"
+    file2_content = "My second document"
+    tar_path = (
+        tmp_path / "test_get_tar_stream.tar.gz"
+        if absolute_path
+        else "test_get_tar_stream.tar.gz"
+    )
+
+    # Create files
+    try:
+        with open(file1_path, "w") as file1:
+            file1.write(file1_content)
+        os.makedirs(dir_path, exist_ok=True)
+        with open(file2_path, "w") as file2:
+            file2.write(file2_content)
```

Comment:
nit: can be simplified to:

```suggestion
        file1.write_text(file1_content)
        dir_path.mkdir(parents=True, exist_ok=True)
        file2.write_text(file2_content)
```